replica/table.cc

```diff
@@ -3440,8 +3440,10 @@ compaction::table_state& compaction_group::as_table_state() const noexcept {
     return *_table_state;
 }
 
-compaction::table_state& table::as_table_state(size_t id) const noexcept {
-    // FIXME: kill it once we're done with all remaining users.
+compaction::table_state& table::try_get_table_state_with_static_sharding(size_t id) const {
```

Comment:
Why do we need `id` to be passed now?
Doesn't it have to be 0 for static sharding?